### ISSUE TITLE
[Feature Request] Smooth cursor

### ISSUE THREAD

**Issue by shaunsingh** [akiyosi/goneovim#224]
Hi, is it a possible to implement a smooth cursor in goneovim? E.g. the cursor smoothly moves/jumps from place to place instead of snaps there? 

For example something like this: https://youtu.be/Yc5cHSZ_9b4. 

A more advanced implementation would be that similar to neovide: 

![](https://github.com/Kethku/neovide/blob/main/assets/AnimatedCursor.gif)

**Comment by akiyosi** [comment]
Yes, I am currently looking into the ability to move the cursor smoothly.
In neovide it's a great feature, but I think what will probably be implemented in goneovim will be a feature with a simpler behavior.

**Comment by JSamir** [comment]
This might sound strange, but beside the fact that smooth cursor just looks better, the inability to use this feature in goneovim makes me use my IDE if I need to look at code with someone else.

The jumping around with shortcuts is impossible for someone to follow if they are not vim users themself (and even then it's hard).

Smooth cursor similar to neovide solves the issue. I do not know how your solution will look like, but from my pov it should not only look good but also solve the issue mentioned here :)

**Comment by Kethku** [comment]
@JSamir I've been thinking about this issue a lot, and was part of the motivation behind the cursor animation in the first place. Animating window transitions was another thing that I think helps in this regard. I do wonder though if there are other features that Neovim guis could build that would help further.

I think this is an underappreciated problem with editors like vim and emacs. Often its so hard to follow unless you're the person driving that people get turned off by it. I'm very interested in ideas for smoothing over that friction even more than just a smooth cursor.

An example: I think minimaps like the one implemented in goneovim help in this regard as it gives viewers a clear picture of where the viewport is in relation to the file overall. I wonder if we can innovate in this area further

**Comment by akiyosi** [comment]
Hi, guys.
This is the current progress, although the Windows build is currently failing.

https://github.com/akiyosi/goneovim/actions/runs/887382694

![May-28-2021-22-25-53](https://user-images.githubusercontent.com/8478977/120057123-b8456c80-c07b-11eb-836d-dad01de58b93.gif)

**Comment by shaunsingh** [comment]
Looks really nice so far! One thing I would add is that vscode changes the speed of the animation based on the distance (the farther it is from the point you want to go, the faster it jumps), which creates a much smoother affect.

**Comment by akiyosi** [comment]
Thanks for the quick comment :D
I think it shouldn't be too hard to implement a speed change based on distance, I'll test it out!

**Comment by akiyosi** [comment]
This is already supported, so it is closed for this issue. The behavior of the smooth cursor and additional feature requests should be discussed in another issue.
